lib/streamlit/runtime/caching/memo_decorator.py

```diff
@@ -220,7 +221,7 @@ def __call__(
         show_spinner: bool = True,
         suppress_st_warning: bool = False,
         max_entries: Optional[int] = None,
-        ttl: Optional[float] = None,
+        ttl: Optional[Union[float, timedelta]] = None,
```

Comment:
It would be better to use type `Optional[Union[float, *int*, timedelta]]`, but doing that leads to strange mypy error:
```Overloaded function implementation does not accept all possible arguments of signature 2```

This looks like related to this discussion https://bugs.python.org/issue42169 , so since it works well with only `float` previously, I decided to add just `timedelta`.

Reply:
Annoyingly, within the python type system, `int` is considered a subclass of `float`. `int` is already part of your signature, by implication.